ably/rest_client_test.go

```diff
@@ -85,58 +87,52 @@ var _ = Describe("RestClient", func() {
 
 	Describe("encoding messages", func() {
 		var (
-			buffer []byte
-			server *httptest.Server
+			buf []byte
+			srv *httptest.Server
+			cli *ably.RestClient
 		)
 
 		BeforeEach(func() {
-			server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			srv = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				var err error
-				buffer, err = ioutil.ReadAll(r.Body)
+				buf, err = ioutil.ReadAll(r.Body)
 				Expect(err).NotTo(HaveOccurred())
 
 				w.WriteHeader(200)
 				w.Header().Set("Content-Type", "application/json")
 				fmt.Fprintf(w, `{}`)
 			}))
+			cli = ably.NewRestClient(testApp.Options())
+			cli.Options.NoTLS = true
+			cli.Options.HTTPClient = newHTTPClientMock(srv)
 		})
 
 		Context("with JSON encoding set up", func() {
 			BeforeEach(func() {
-				testParamsCopy := testApp.Options
-				testParamsCopy.Protocol = ably.ProtocolJSON
-				client = ably.NewRestClient(testParamsCopy)
-
-				client.RestEndpoint = strings.Replace(client.RestEndpoint, "https", "http", 1)
-				client.HTTPClient = createMockedRestClient(server)
+				cli.Options.Protocol = ably.ProtocolJSON
 
-				err := client.Channel("test").Publish("ping", "pong")
+				err := cli.Channel("test").Publish("ping", "pong")
 				Expect(err).NotTo(HaveOccurred())
 			})
 
 			It("encode the body of the message in JSON", func() {
 				var anyJson []map[string]interface{}
-				err := json.Unmarshal(buffer, &anyJson)
+				err := json.Unmarshal(buf, &anyJson)
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
 		Context("with msgpack encoding set up", func() {
 			BeforeEach(func() {
-				testParamsCopy := testApp.Options
-				testParamsCopy.Protocol = ably.ProtocolMsgPack
-				client = ably.NewRestClient(testParamsCopy)
-
-				client.RestEndpoint = strings.Replace(client.RestEndpoint, "https", "http", 1)
-				client.HTTPClient = createMockedRestClient(server)
+				cli.Options.Protocol = ably.ProtocolMsgPack
 
-				err := client.Channel("test").Publish("ping", "pong")
+				err := cli.Channel("test").Publish("ping", "pong")
 				Expect(err).NotTo(HaveOccurred())
 			})
 
 			It("encode the body of the message using msgpack", func() {
 				var anyMsgPack []map[string]interface{}
-				err := msgpack.Unmarshal(buffer, &anyMsgPack)
+				err := msgpack.Unmarshal(buf, &anyMsgPack)
```

Comment:
As above, `buf` and `cli` instead of `buffer` and `client` IMHO does not make this code more readable.